internal/pkg/manifest/union.go

```diff
@@ -0,0 +1,156 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"gopkg.in/yaml.v3"
+)
+
+// Union is a type used for yaml keys that may be of type Simple or Advanced.
```

Comment:
Let me know if I'm being pedantic (requiring precision without illuminating anything 😂). A `Union` to me sounds like an _inclusive or_ where it's valid to set both `Simple` or `Advanced` or both.
Whereas this type is an `xor` operation.

What do you think about the following names:
```go
type XOrConfig[Basic, Advanced any] struct { }
type EitherConfig[Basic, Advanced any] struct { }
type LeveledConfig[Basic, Advanced any] struct { }
type ProgressiveConfig[Basic, Advanced any] struct { }
```

Reply:
i think a union type implies that it's not valid to be both at once 🥺: 
> In contrast with a [record](https://en.wikipedia.org/wiki/Record_(computer_science)) (or structure), which could be defined to contain a float and an integer; in a union, there is only one value at any given time.

(https://en.wikipedia.org/wiki/Union_type)